pkg/apis/core/types_shootextensionstatus.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShootExtensionStatus holds the status information of extensions of a Shoot cluster
+type ShootExtensionStatus struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	// Designed to have an owner reference to the associated Shoot resource
+	metav1.ObjectMeta
+	// Specification of the ShootExtensionStatus.
+	Spec ShootExtensionStatusSpec
```

Comment:
WDYT about adding not using `Spec` but `ExtensionStatuses` directly here? I think it's somehow confusing to store a state (`ProviderStatus`) in a `Spec` field. In addition, there is no controller which would use this `Spec` and transfer it to an actual state, i.e., there is no `Spec <-> Status` pair. (/cc @rfranzke)

Reply:
Ok I see what you mean.
I modeled it after the ShootState resource as it is very similar mechanics.
So the argument to model it like this is only for consistency.
WDYT @timuthy

Reply:
Yes, I kind of wondered why we did it for the `ShootState` in the first place and maybe should change it there after some time.

I get the consistency argument but rather vote for following the K8s convention (see [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status)) if we now have the chance to do so:

> All objects that represent a physical resource whose state may vary from the user's desired intent SHOULD have a "spec" and a "status". Objects whose state cannot vary from the user's desired intent MAY have only "spec", and MAY rename "spec" to a more appropriate name.

Reply:
I'd also vote for @timuthy's proposal. When we promote the `ShootState` to `v1beta1` we should do it similarly 👍🏻

Reply:
thanks for your input. Will adapt it